lib/live_view_native/swiftui/rules_parser/tokens.ex

```diff
@@ -123,101 +105,67 @@ defmodule LiveViewNative.SwiftUI.RulesParser.Tokens do
-    |> ascii_string([?a..?z, ?A..?Z, ?0..?9, ?_], min: 0)
-    |> reduce({Enum, :join, [""]})
-  end
-
-  def enclosed(start \\ empty(), open, combinator, close) do
-    start
-    |> ignore_whitespace()
-    |> ignore(string(open))
-    |> ignore_whitespace()
-    |> concat(combinator)
-    |> ignore_whitespace()
-    |> ignore(string(close))
-    |> ignore_whitespace()
-  end
-
-  #
-  # Collections
-  #
-
-  def comma_separated_list(combinator \\ empty(), elem_combinator) do
-    delimiter_separated_list(combinator, elem_combinator, ",", true)
-  end
-
-  def non_empty_comma_separated_list(combinator, elem_combinator) do
-    delimiter_separated_list(combinator, elem_combinator, ",", false)
-  end
-
-  def delimiter_separated_list(combinator, elem_combinator, delimiter, allow_empty \\ true) do
-    #  1+ elems
-    non_empty =
-      elem_combinator
-      |> repeat(
-        ignore_whitespace()
-        |> ignore(string(delimiter))
-        |> ignore_whitespace()
-        |> concat(elem_combinator)
+  def variable(opts \\ []) do
+    expected(
+      ascii_string([?a..?z, ?A..?Z, ?_], 1)
+      |> ascii_string([?a..?z, ?A..?Z, ?0..?9, ?_], min: 0)
+      |> reduce({Enum, :join, [""]})
+      |> post_traverse({PostProcessors, :to_elixir_variable_ast, []}),
+      Keyword.merge(
+        [error_message: "expected a variable"],
+        opts
       )
-
-    empty_ = ignore_whitespace(empty())
-
-    if allow_empty do
-      combinator
-      |> choice([non_empty, empty_])
-    else
-      combinator
-      |> concat(non_empty)
-    end
+    )
   end
 
-  def newline_separated_list(elem_combinator) do
-    #  1+ elems
-    ignore_whitespace()
-    |> concat(elem_combinator)
-    |> repeat(
-      choice([
-        ignore(optional(whitespace_except("\n", min: 1)))
-        |> ignore(string("\n"))
-        |> ignore(whitespace(min: 1))
-        |> concat(elem_combinator),
-        # Require at least one whitespace
-        ignore(whitespace(min: 1))
-      ])
+  def modifier_name() do
+    expected(
+      ascii_string([?a..?z, ?A..?Z, ?_], 1)
+      |> ascii_string([?a..?z, ?A..?Z, ?0..?9, ?_], min: 0)
+      |> reduce({Enum, :join, [""]}),
+      error_message: "Expected a modifier name",
+      error_parser:
+        choice([
+          non_whitespace(also_ignore: String.to_charlist("[](),"), fail_if_empty: true),
+          non_whitespace(also_ignore: String.to_charlist("]),"), fail_if_empty: true),
+          non_whitespace()
+        ]),
+      show_incorrect_text?: true
     )
   end
 
-  def key_value_pair() do
-    ignore_whitespace()
-    |> concat(word())
-    |> concat(ignore(string(":")))
-    |> ignore_whitespace()
-    |> concat(
-      choice([
-        literal(),
-        parsec(:attr),
-        parsec(:event),
-        parsec(:ime),
-        parsec(:helper_function),
-        parsec(:nested_attribute),
-        parsec(:key_value_list),
-        variable()
-      ])
+  def module_name() do
+    expected(
```

Comment:
I suspect this combinator is being used for `Color.red` referencing `Color` as a "module", if so that isn't going to be the correct term on the Swift side. In this particular case `Color` is a type and I suspect in nearly all cases we'd be referencing types but I'd want @carson-katri to confirm/deny

Reply:
I've renamed it to `type_name`